Behrooz v2

Behrooz-Q opened the pull request:
Similar to last PR, except:
- One commit for each major change in Solr 
- One commit for all the files in app

Behrooz-Q committed:
added solr

Behrooz-Q committed:
removed unnecessary files from solr

Behrooz-Q committed:
added plugins

Behrooz-Q committed:
added solr core

Behrooz-Q committed:
added app

stephhazlitt commented:
@Behrooz-Q this is looking great. Thank you for taking the time to commit in steps.

A few things to do before we merge this PR:

- for all of your `/app` and original code files, can you insert the Apache 2.0 header (see example below)
- can you add the citation and licence-details that allow the copy+paste use of`solr` into the README, even if just a quick draft for now

Once these are done let's merge and you can add details to the README and final report copies etc. in subsequent PRs.


Apache 2.0 header for all _your_ code files:

```
# Copyright 2023 Province of British Columbia
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and limitations under the License.
```

stephhazlitt reviewed (changes requested):
LGTM with the changes/suggestions added via comment.

Behrooz-Q committed:
added solr licenses folder

Behrooz-Q committed:
adding Apache 2.0 header to Funcs.py, Models.py, Variables.py, and search_engine.py

Behrooz-Q committed:
modified README.md, adding citation and license-details for Solr and Vector Scoring Plugin for Solr

Behrooz-Q committed:
add README_files folder to gitignore

Behrooz-Q committed:
modified dependencies section in README.md

Behrooz-Q commented:
Hi Stephanie, Thank you for your valuable feedback. Based on your comments, I've made several improvements to the repository:
1. Added a dependencies section in README.md with links to the Solr website and its license.txt.
2. Included a license folder for Solr to ensure proper compliance.
3. Provided a link to the Vector Scoring Plugin for Solr repository.
4.Added the license header to all four .py files in the /app directory.

stephhazlitt commented:
LGTM. Feel free to merge this and use a follow-up PR for any final details in the `README` and/or committing the project report or slides (PDF formats) etc. 

IMO, a finished `README` should include details on what software is needed to execute the app, order of scripts to be run (if manual) and any other relevant details for a user to get up and running.

Behrooz-Q committed:
added environment.yml file

Behrooz-Q committed:
modified README.md file, added 'problem statement' and 'how to run' sectinos
